hvpy/api.py

```diff
@@ -1,13 +1,76 @@
 __all__ = ["fake_api_call"]
 
+from datetime import datetime
+from typing import Optional
+import requests
+from pydantic import BaseModel, Field
 
-def fake_api_call():
+
+class InputParameters(BaseModel):
+    date: datetime
+    sourceId: int
+    jpip: Optional[bool] = False
+    Json: Optional[bool] = Field(False, alias="json")
+
+
+def parse_response(response: requests.Response, output_parameters: str):
     """
     _summary_
 
+    Parameters
+    ----------
+    response: ?
+        _description_
+    output_parameters: ?
+        _description_
+
     Returns
     -------
     _type_
         _description_
     """
-    return {}
+    if output_parameters == "binary":
+        return response.content
+    elif output_parameters == "url":
+        return response.url
+    elif output_parameters == "json":
+        return response.json()
+
+
+def execute_api_call(url: str, input_parameters: dict, output_parameters: str):
+    """
+    _summary_
+
+    Parameters
+    ----------
+    url: str
+        _description_
+    input_parameters: dict
+        _description_
+    output_parameters: ?
+        _description_
```

Comment:
From the `HvpyParameters` class do you mean the `<endpoint>InputParameters` class?

Reply:
HvpyParameters will define the dict() function, and it also implements the default get_output_type() function.
So for any API endpoint that inherits HvpyParameters we don't need to redefine dict(). And we don't need to redefine get_output_type unless it's different from OutputType.Raw or unless it changes based on input parameters.
Otherwise most Parameters will just inherit it and just need to define the fields with no extra code.